### PR TITLE
*: support auto tls on client side

### LINKED ISSUE #5149: auto-tls on client port
When users do not provide key/cert but provide client-auto-tls flag, etcd can generate self signed cert to serve secure connection. This will make etcd easier to be secure. The end goal is to make etcd secure by default. This is a step forward.

/cc @heyitsanthony

### PR DESCRIPTION
**Pull request by xiang90**
Fix #5149

/cc @heyitsanthony

### CODE REVIEW

File: Documentation/op-guide/configuration.md [hunk 1]
```diff
@@ -201,6 +201,11 @@ The security flags help to [build a secure etcd cluster][security].
 + default: none
 + env variable: ETCD_TRUSTED_CA_FILE
 
+### --auto-tls
```
Comment (heyitsanthony) [inline Documentation/op-guide/configuration.md]: should this be `--client-auto-tls` to be consistent with the `--peer-auto-tls` option?

Comment (xiang90) [inline Documentation/op-guide/configuration.md]: for all client tls options, we do not have client prefix.

Comment (heyitsanthony) [inline Documentation/op-guide/configuration.md]: ok

**Comment by heyitsanthony** [comment]
lgtm but should probably have an e2e test?

**Comment by xiang90** [comment]
@heyitsanthony Will add e2e test.

**Comment by xiang90** [comment]
@heyitsanthony e2e tests added. PTAL.

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**